sympy/ntheory/tests/test_generate.py

```diff
@@ -244,3 +244,12 @@ def test_sieve_iter():
 def test_sieve_repr():
     assert "sieve" in repr(sieve)
     assert "prime" in repr(sieve)
+
+def test_nprime():
+    assert nprimes(10) == [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
+    assert nprimes(13, 15) == [41, 43, 47]
+    raises(ValueError, lambda: nprimes(-1))
+    raises(ValueError, lambda: nprimes(0))
+    raises(ValueError, lambda: nprimes(10, 0))
+    raises(ValueError, lambda: nprimes(10,-5))
+    assert nprimes(10, 2) == []
```

Comment:
since this is not working like a `range` function I think we can be more strict in requiring that `k >=n` .

Also, I like the name `primes` better.

Reply:
@smichr `nprimes` seems to be better.

Reply:
The `n` does not provide more clarity than the pluralizing `s`. I prefer one of the two options I described in the issue and am barely +1/2 because there is already a clear way to do this with `primerange`. But something that makes me close to +1 is getting a list rather than a generator: `list(primerange(prime(10)+1))` vs `primes(10)`.

Reply:
I'll make it primes